### PR TITLE
Fix for Issue 1290

### PR DESCRIPTION
**Pull request by momuno**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [x] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)
https://github.com/Azure/azure-iot-sdk-c/issues/1290
# Description of the problem
Addresses Issue 1290

# Description of the solution
Uses the unused error `IOTHUB_MESSAGE_INVALID_TYPE` to differentiate when the ascii check fails in calling `IoTHubMessage_SetProperty.`
Adds 2 unit tests accordingly.

This PR comes from a branch of azure-iot-sdk-c instead of from a fork of that repo.  This is an attempt to pass the esp testing that was failing (possibly due to "References service endpoint. PRs from repository forks are not allowed to access secrets in the pipeline").

### CODE REVIEW

**Review by danewalton — APPROVED** [review]

File: iothub_client/inc/iothub_message.h [hunk 1]
```diff
@@ -191,9 +191,9 @@ MOCKABLE_FUNCTION(, MAP_HANDLE, IoTHubMessage_Properties, IOTHUB_MESSAGE_HANDLE,
 *
 * @param   iotHubMessageHandle Handle to the message.
 *
-* @param   key name of the property to set.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
+* @param   key name of the property to set using valid ASCII characters.  Note that when sending messages via the HTTP transport, this value must not contain spaces.
```
Comment (ericwol-msft) [inline iothub_client/inc/iothub_message.h:194]: this is not true, it's defined as below

Property names and values can only contain ASCII **alphanumeric** characters, plus {'!', '#', '$', '%, '&', ''', '*', '+', '-', '.', '^', '_', '`', '|', '~'} 

https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct

Comment (momuno) [reply iothub_client/inc/iothub_message.h:194]: How do you think I best explain that here?  I was hoping "valid" would suffice.  I could just say between dec value [32, 126]

Comment (momuno) [reply iothub_client/inc/iothub_message.h:194]: or [' ', '~']

Comment (momuno) [reply iothub_client/inc/iothub_message.h:194]: Except this page says there should be no spaces.... https://docs.microsoft.com/en-us/azure/iot-hub/iot-c-sdk-ref/iothub-message-h/iothubmessage-setproperty

Comment (ericwol-msft) [reply iothub_client/inc/iothub_message.h:194]: a space is not an ASCII alphanumeric character

Comment (momuno) [reply iothub_client/inc/iothub_message.h]: The docs specify no space for the key.  But no mention for the value.
The code allows spaces.  Are you saying we should not allow spaces for the value either?

File: iothub_client/src/iothub_message.c [hunk 2]
```diff
@@ -43,8 +43,8 @@ static bool ContainsOnlyUsAscii(const char* asciiValue)
     const char* iterator = asciiValue;
     while (iterator != NULL && *iterator != '\0')
     {
-        // Allow only printable ascii char
-        if (*iterator < ' ' || *iterator > '~')
+        // Allow only printable ascii char (excluding space ' ').
```
Comment (ericwol-msft) [inline iothub_client/src/iothub_message.c:46]: This is still not right. These chars are NOT allowed
" ( ) , / : ; < = > ? @ { }

Comment (momuno) [reply iothub_client/src/iothub_message.c:46]: I love how off the original code was from the specs lol.

**Comment by momuno** [comment]
@wesmc7777 is updating the iothub_message.h file for documentation so I have removed that change from this commit.

File: iothub_client/src/iothub_message.c (hunk 2 above)
Comment (momuno) [reply iothub_client/src/iothub_message.c]: Due to validation results, we should leave this as is.  Some protocols allows all characters [dec 32, dec 126], while others are limited.  It is also c2d vs d2c dependent.  I do not know of a way the code can determine which protocol is being used at this level, and I do not think it should be responsible to.  This puts the onus on the IoT Hub documentation to inform the user, and if the user uses invalid ascii characters, they will encounter an error elsewhere.

**Pull request merged**